src/utils/serial.rs

```diff
@@ -48,6 +48,13 @@ impl From<Serial> for u32 {
     }
 }
 
+impl Serial {
+    /// Checks if a serial was generated after or is equal to another given serial
+    pub fn is_no_older_than(&self, other: &Serial) -> bool {
+        other.0 <= self.0
+    }
+}
```

Comment:
But isn't `PartialOrd` is basically this, since `Serial` wrap arounds?

Reply:
Right, I personally like the named function more though, as it is more explicit in code. Or do you feel like the function should make an attempt to handle the wrapping somehow?

Reply:
done